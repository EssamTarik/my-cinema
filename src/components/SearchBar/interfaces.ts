export interface IProps {
  onChange: (text: string) => void;
  value: string;
}