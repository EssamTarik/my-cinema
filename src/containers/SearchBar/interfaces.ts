export interface IProps {
  searchMovies: (query: string, page: number) => void;
}