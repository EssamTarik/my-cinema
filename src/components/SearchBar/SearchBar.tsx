import React from "react";
import { useIntl } from "react-intl";
import Form from "react-bootstrap/Form";
import { IProps } from "./interfaces";

const SearchBar = ({ onChange, value = "" }: IProps) => {
  const intl = useIntl();
  return (
    <Form.Control
      value={value}
      placeholder={intl.formatMessage({ id: "searchBar.placeholder" })}
      type="text"
      onChange={({ target: { value: newValue } }) => onChange(newValue)}
    />
  );
};

export default SearchBar;
