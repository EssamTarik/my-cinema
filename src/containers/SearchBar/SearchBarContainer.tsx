import React, { useCallback } from 'react';
import debounce from 'lodash.debounce';
import { SearchBar } from '../../components';
import { IProps } from './interfaces';

const SearchBarContainer = ({ searchMovies }: IProps) => {
  const handleChange = debounce(useCallback((query: string) => searchMovies(query, 1), [searchMovies]), 300);

  return <SearchBar onChange={handleChange} />
}

export default SearchBarContainer;