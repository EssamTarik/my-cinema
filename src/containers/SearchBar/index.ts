import { connect } from 'react-redux';
import { searchMovies } from '../../store/movies/actions';
import SearchBarContainer from './SearchBarContainer';

const mapDispatchToProps = {
  searchMovies
}

export default connect(null, mapDispatchToProps)(SearchBarContainer);