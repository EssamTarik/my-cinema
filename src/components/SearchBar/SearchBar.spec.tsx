jest.mock('react-intl', () => ({ useIntl: () => ({ formatMessage: () => '' }) }))
import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";

import SearchBar from "./SearchBar";

describe('SearchBar', () => {
  afterEach(cleanup);
  const mockOnChange = jest.fn();

  it("renders", () => {
    const { asFragment } = render(
      <SearchBar onChange={mockOnChange} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Calls on change", () => {
    const value = 'test'
    const { container } = render(
      <SearchBar onChange={mockOnChange} />
    );

    const input = container.querySelector('input');
    fireEvent.change(input!, { target: { value } })
    expect(mockOnChange).toHaveBeenCalledWith(value)
  });
})

