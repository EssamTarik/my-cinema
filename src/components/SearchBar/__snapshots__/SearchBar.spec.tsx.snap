// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders 1`] = `
<DocumentFragment>
  <input
    class="form-control"
    placeholder=""
    type="text"
    value=""
  />
</DocumentFragment>
`;
